src/util.ts

```diff
@@ -30,90 +30,102 @@ import type { TomlDate } from './date.js'
 import { TomlError } from './error.js'
 
 export type TomlPrimitive =
-	| string
-	| number
-	| boolean
-	| TomlDate
-	| { [key: string]: TomlPrimitive }
-	| TomlPrimitive[]
-
-export function indexOfNewline (str: string, start = 0, end = str.length) {
-	let idx = str.indexOf('\n', start)
-	if (str[idx - 1] === '\r') idx--
-	return idx <= end ? idx : -1
+  | string
+  | number
+  | BigInt
```

Comment:
[Don't use boxed types](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#number-string-boolean-symbol-and-object). The doc doesn't mention BigInt but it also applies to it. Use lowercase `bigint`.